### ISSUE TITLE
SSL infinity recursion error when using this with gevent.

### ISSUE THREAD

**Issue by luxrck** [shazow/urllib3#1176]
#### reproduce
```python
import gevent.monkey
from requests.packages.urllib3.util.ssl_ import create_urllib3_context

gevent.monkey.patch_all()
create_urllib3_context()
```

**Comment by Lukasa** [comment]
Sorry, but I can't reproduce: running your code on my machine produces no problem whatsoever. Can you provide the output of `pip freeze` and `python --version`?

**Comment by luxrck** [comment]
os: archlinux
python: 3.6.1
gevent: 1.2.1
requests: 2.14.2

**Comment by Lukasa** [comment]
Based on the fix you provided in #1177, it seems that gevent is monkeypatching the `SSLContext` object. It is *vitally important* when you use gevent that you apply the monkeypatch *before* you import any other code. Code frequently will use `from` imports to bring objects into their own namespaces, and monkeypatching after doing that will cause code to break.

Rather than force every library in the world to bend to gevent's will, just re-order your code:

```python
import gevent.monkey
gevent.monkey.patch_all()

from requests.packages.urllib3.util.ssl_ import create_urllib3_context
create_urllib3_context()
```

This will resolve your issues nicely.

**Comment by luxrck** [comment]
sure it is.
